worldengine/cli/main.py

```diff
@@ -400,79 +403,78 @@ def main():
-                               step, options.ocean_level, world_format,
-                               options.verbose, black_and_white=options.black_and_white)
+        world = generate_world(world_name, args.width, args.height,
+                               seed, args.number_of_plates, args.output_dir,
+                               step, args.ocean_level, world_format,
+                               args.verbose, black_and_white=args.black_and_white)
         if produce_grayscale_heightmap:
             generate_grayscale_heightmap(world, produce_grayscale_heightmap)
-        if options.rivers_map:
-            generate_rivers_map(world, options.rivers_map)
+        if args.rivers_map:
+            generate_rivers_map(world, args.rivers_map)
 
     elif operation == 'plates':
         print('')  # empty line
         print('starting (it could take a few minutes) ...')
 
-        generate_plates(seed, world_name, options.output_dir, options.width,
-                        options.height, num_plates=number_of_plates)
+        generate_plates(seed, world_name, args.output_dir, args.width,
+                        args.height, num_plates=args.number_of_plates)
 
     elif operation == 'ancient_map':
         print('')  # empty line
         print('starting (it could take a few minutes) ...')
 
         # First, some error checking
-        if options.sea_color == "blue":
+        if args.sea_color == "blue":
             sea_color = (142, 162, 179, 255)
-        elif options.sea_color == "brown":
+        elif args.sea_color == "brown":
             sea_color = (212, 198, 169, 255)
         else:
-            usage("Unknown sea color: " + args[
-                0] + "  Select from [" + SEA_COLORS + "]")
-        if not options.world_file:
+            usage("Unknown sea color: " + args.sea_color +
+                  " Select from [" + SEA_COLORS + "]")
+        if not args.world_file:
             usage(
                 "For generating an ancient map is necessary to specify the " +
                 "world to be used (-w option)")
-        world = load_world(options.world_file)
+        world = load_world(args.world_file)
 
         print_verbose(" * world loaded")
 
-        if not options.generated_file:
-            options.generated_file = "ancient_map_%s.png" % world.name
-        operation_ancient_map(world, options.generated_file,
-                              options.resize_factor, sea_color,
-                              options.draw_biome, options.draw_rivers,
-                              options.draw_mountains, options.draw_outer_border)
+        if not args.generated_file:
+            args.generated_file = "ancient_map_%s.png" % world.name
+        operation_ancient_map(world, args.generated_file,
+                              args.resize_factor, args.sea_color,
+                              args.draw_biome, args.draw_rivers,
+                              args.draw_mountains, args.draw_outer_border)
```

Comment:
Thanks for refactoring all of this! Maybe later we could split this function in argument processing, validation and execution